### PR TITLE
Fix compiled mode via pull_value!

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
See the discussion in https://github.com/JuliaDiff/ReverseDiff.jl/issues/143

### CODE REVIEW

File: test/local_sensitivity/sde.jl [hunk 1]
```diff
@@ -37,13 +37,11 @@ p2 = [1.01,0.87]
   σ_oop_linear(u,p,t) = p[2]*u
 
   function f_oop_linear(u::Tracker.TrackedArray,p,t)
-    dx = p[1]*u[1]
-    Tracker.collect([dx])
+    p[1].*u
```
Comment (ChrisRackauckas) [inline test/local_sensitivity/sde.jl:40]: @frankschae this decreases it a ton and now the SO goes away. Not really the best long term solution, but it's at least an improvement we should do.

**Comment by ChrisRackauckas** [comment]
Why does the first `@info` in the SDE tests give a different value? @frankschae

**Comment by frankschae** [comment]
The first one is a comparison to the ODE case, i.e., the parameter in front of the diffusion term is set to zero -- thought this might be useful to check if something happens just because of the SDE part.

File: test/local_sensitivity/sde.jl (hunk 1 above)
Comment (frankschae) [reply test/local_sensitivity/sde.jl:40]: Oh, I didn't know that this works. I always used Tracker with the collect so far.

**Comment by ChrisRackauckas** [comment]
Interesting okay, we should probably note that somewhere.

**Pull request merged**